File: app/src/main/java/tw/com/louis383/coffeefinder/CoffeeTripApplication.java
package tw.com.louis383.coffeefinder;

import android.app.Application;
import android.support.v7.app.AppCompatDelegate;

import tw.com.louis383.coffeefinder.di.AppComponent;
import tw.com.louis383.coffeefinder.di.AppModule;
import tw.com.louis383.coffeefinder.di.DaggerAppComponent;

/**
 * Created by louis383 on 2017/1/24.
 */

public class CoffeeTripApplication extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);

        appComponent = DaggerAppComponent.builder().appModule(new AppModule(this)).build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/tw/com/louis383/coffeefinder/CoffeeTripApplication.kt
package tw.com.louis383.coffeefinder

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import com.google.android.material.color.DynamicColors
import dagger.hilt.android.HiltAndroidApp
import tw.com.louis383.coffeefinder.utils.QuickCheckUtils

/**
 * Created by louis383 on 2017/1/24.
 */
@HiltAndroidApp
class CoffeeTripApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        if (QuickCheckUtils.canApplyDynamicColor()) {
            DynamicColors.applyToActivitiesIfAvailable(this)
        }
    }
}